src/node_url.cc

```diff
@@ -1306,6 +1308,7 @@ namespace url {
           args[3]->IsObject());
     CHECK(args[4]->IsFunction());
     Utf8Value input(env->isolate(), args[0]);
+    CHECK(!input.IsInvalidated());
```

Comment:
If you remove the `CHECK(args[0]->IsString());` (which I agree makes sense), can this be turned into just returning instead of crashing?

ditto for the remainder of the file (or am I missing something here?)